src/app/modules/workspace/__tests__/workspace.service.spec.ts

```diff
@@ -275,4 +278,181 @@ describe('workspace.service', () => {
       expect(actual._unsafeUnwrapErr()).toBeInstanceOf(WorkspaceNotFoundError)
     })
   })
+
+  describe('deleteWorkspace', () => {
+    const mockFormId = new ObjectId()
+    const mockWorkspace = {
+      _id: new ObjectId(),
+      admin: new ObjectId(),
+      title: 'workspace1',
+      formIds: [mockFormId],
+      count: 0,
+    }
+
+    it('should return undefined when successfully deleted workspace', async () => {
```

Comment:
usually we should think carefully about returning `undefined`, as it's an uninformative response type. one alternative could be to use `findOneAndRemove` in the db layer so as to be able to at least return the deleted document.

Reply:
Updated to use `findOneAndRemove`!